gensim/models/phrases.py

```diff
@@ -115,7 +115,7 @@ def _is_single(obj):
         return True, obj_iter
     else:
         # If the first item isn't a string, assume obj is a corpus
-        return False, obj_iter
+        return False, list(obj_iter)
```

Comment:
It also works if I return just "obj" instead of "list(obj_iter)" but it fails if the obj is the an iterator such as " iter([ [] , [] ]) " , so I am thinking of adding a special case to handle such iterator and else otherwise just return the "obj".

Reply:
@menshikh-iv  What do you think about the above proposed changes ?

Reply:
@sj29-innovate anyway, `list(...)` unacceptable here.

Reply:
Yes , I agree that is not a very good way to do that . This works for me without using "list(...)" ,
```
    obj_iter = iter(obj)
    temp_iter = obj_iter
    try:
        peek = next(obj_iter)
        obj_iter = it.chain([peek], obj_iter)
    except StopIteration:
        # An empty object is a single document
        return True, obj
    if isinstance(peek, string_types):
        # It's a document, return the iterator
        return True, obj_iter
    if temp_iter == obj :
        #Checking for iterator to the object    
    	return False , obj_iter
    else:
        # If the first item isn't a string, assume obj is a corpus
        return False, obj
'''
I realised there was no use of using the "list(...)" , works well with the above change.